src/core/sagas/user.js

```diff
@@ -0,0 +1,34 @@
+// Disabled because of
+// https://github.com/benmosher/eslint-plugin-import/issues/793
+/* eslint-disable import/order */
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+/* eslint-enable import/order */
+
+import { SET_AUTH_TOKEN } from 'core/constants';
+import log from 'core/logger';
+import { getState } from 'core/sagas/utils';
+import { userProfileLoaded } from 'core/reducers/user';
+import { userProfile as userProfileApi } from 'core/api';
+
+export function* fetchUserProfile({ payload }) {
+  const { token } = payload;
+
+  try {
+    const state = yield select(getState);
+
+    const response = yield call(userProfileApi, {
+      api: {
+        ...state.api,
+        token,
+      },
+    });
+
+    yield put(userProfileLoaded({ profile: response }));
+  } catch (error) {
+    log.warn(`Current user profile failed to load: ${error}`);
+  }
+}
+
+export default function* userSaga() {
+  yield takeLatest(SET_AUTH_TOKEN, fetchUserProfile);
```

Comment:
We need the `username` in the header for instance, so I think we'll need to fetch the user profile anyway. I chose this action because it seems to be done only once on first hit (+ page reloads).